src/mcp/shared/metadata_utils.py

```diff
@@ -0,0 +1,45 @@
+"""Utility functions for working with metadata in MCP types.
+
+These utilities are primarily intended for client-side usage to properly display
+human-readable names in user interfaces.
```

Comment:
Maybe even including spec link

```suggestion
These utilities are primarily intended for client-side usage to display
human-readable names in user interfaces in a spec compliant way
```